src/main/java/core/basesyntax/service/ReportToCsv.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.StockDao;
+import core.basesyntax.db.StockDaoStorageImpl;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class ReportToCsv implements Report {
+    private static final String FILE_NAME = "report.csv";
+    private final StockDao stockDao = new StockDaoStorageImpl();
+
+    @Override
+    public void prepare() {
+        try {
+            BufferedWriter writer = new BufferedWriter(new FileWriter(FILE_NAME));
+
+            for (String product : stockDao.getProductsList()) {
+                writer.write(product + "," + stockDao.get(product) + System.lineSeparator());
+            }
+
+            writer.close();
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);

```

Comment:
While rethrowing the `IOException` as a `RuntimeException` is one way to handle it, consider providing a more descriptive error message to aid in debugging. This can help identify the source of the error more easily.